### PR TITLE
[MXNET-628] Fix example on text classification using LSTMs on IMDB dataset

### PR DESCRIPTION
**Pull request by anirudhacharya**
## Description ##
Add initializer and callback and tune hyperparameters to fix the example implementation of a text classification model using LSTMs on IMDB dataset.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
- @hetong007 @ankkhedia

### CODE REVIEW

**Comment by nswamy** [comment]
are there tests for these? if not could you add some tests that run on CI?

File: example/rnn/bucket_R/aclImdb_lstm_classification.R [hunk 1]
```diff
@@ -40,20 +40,31 @@ optimizer <- mx.opt.create("adadelta", rho = 0.92, epsilon = 1e-06, wd = 2e-04,
...
-                          batch.end.callback = NULL, 
-                          epoch.end.callback = epoch.end.callback)
+                                         train.data = train.data,
+                                         eval.data = eval.data,
+                                         num.round = num.round,
+                                         ctx = devices,
+                                         verbose = FALSE,
+                                         metric = mx.metric.accuracy,
+                                         optimizer = optimizer,
+                                         initializer = mx.init.Xavier(rnd_type = "gaussian", factor_type = "in", magnitude = 2),
```
Comment (hetong007) [inline example/rnn/bucket_R/aclImdb_lstm_classification.R]: Is this line too long?

File: example/rnn/bucket_R/aclImdb_lstm_classification.R [hunk 2]
```diff
@@ -40,20 +40,31 @@ optimizer <- mx.opt.create("adadelta", rho = 0.92, epsilon = 1e-06, wd = 2e-04,
...
-            ignore_label = -1, loss_output = "softmax",
-            output_last_state = F, masking = T)
+  rnn.graph(config = "seq-to-one",
+            cell_type = "lstm",
+            num_rnn_layer = 1,
+            num_embed = 2,
+            num_hidden = 6,
+            num_decode = 2,
+            input_size = vocab,
+            dropout = 0.2,
```
Comment (hetong007) [inline example/rnn/bucket_R/aclImdb_lstm_classification.R]: Can you please mention any results or reference to other implementions on this change?

**Comment by anirudhacharya** [comment]
@nswamy I am in the process of adding all R tutorials to run as part of the Nightly Tests. I can add a test for this there. Because training this model takes a considerable amount of time and adding tests for this example on the CI will increase the CI runtime considerably.

File: example/rnn/bucket_R/aclImdb_lstm_classification.R (hunk 1 above)
Comment (anirudhacharya) [reply example/rnn/bucket_R/aclImdb_lstm_classification.R]: it did not break any lint check, but i will split it up into different lines.

File: example/rnn/bucket_R/aclImdb_lstm_classification.R (hunk 2 above)
Comment (anirudhacharya) [reply example/rnn/bucket_R/aclImdb_lstm_classification.R]: the previous dropout rate of 0.5 would give ~0.81 accuracy on training data and ~0.8114 on test data. 
A dropout rate of 0.2 gives ~0.84 on training data and ~0.8119 on test data.
I just went around tweaking the values by intuition, there was no specific reference I was looking at.

I will add this as a comment?

Comment (hetong007) [reply example/rnn/bucket_R/aclImdb_lstm_classification.R]: It'd be better if a comment with expected results is added to somewhere in the code. Mentioning the previous result with `dropout = 0.5` is not necessary.

Comment (anirudhacharya) [reply example/rnn/bucket_R/aclImdb_lstm_classification.R]: Makes sense. Have added it.

**Comment by hetong007** [comment]
LGTM.

**Comment by anirudhacharya** [comment]
@hetong007 please merge it.
